src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1192 @@ const ActorInit En_Niw_InitVars = {
+    }
+
+    sq = 20.0f;
+
+    if (this->unk_2A8 != 0 && this->actor.xyzDistFromLinkSq < SQ(sq) && player->invincibilityTimer == 0) {
+        func_8002F6D4(globalCtx, &this->actor, 2.0f, this->actor.posRot.rot.y, 0.0f, 0x10);
+    }
+
+    func_80AB747C(&this->actor, globalCtx);
+
+    if (this->sfxTimer2 == 0 && this->actionFunc == func_80AB6BF8) {
+        this->sfxTimer2 = 7;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKU_WAKEUP);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB5BF8.s")
+    if (this->sfxTimer1 == 0) {
+        if (this->actionFunc != func_80AB6570) {
+            this->sfxTimer1 = 30;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_A);
+        } else {
+            this->sfxTimer1 = 300;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_N);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB5FA0.s")
+    if (this->unk_2A8 == 0) {
+        Collider_CylinderUpdate(&this->actor, &this->collider);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6100.s")
+        if (this->actor.params != 0x000A && this->actor.params != 0x000D && this->actor.params != 0x000E &&
+            this->actor.params != 0x0004) {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB627C.s")
+        if (this->actionFunc != func_80AB6BF8 && this->actionFunc != func_80AB6D08 &&
+            this->actionFunc != func_80AB6324 && this->actionFunc != func_80AB63A8 &&
+            this->actionFunc != func_80AB6450) {
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6324.s")
+s32 EnNiw_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnNiw* this = THIS;
+    Vec3f D_80AB8700 = { 0.0f, 0.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB63A8.s")
+    if (limbIndex == 13) {
+        rot->y += (s16)this->unk_2DC;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6450.s")
+    if (limbIndex == 15) {
+        rot->y += (s16)this->unk_2E0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6570.s")
+    if (limbIndex == 11) {
+        rot->x += (s16)this->unk_2D8;
+        rot->y += (s16)this->unk_2D4;
+        rot->z += (s16)this->unk_2D0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6A38.s")
+    if (limbIndex == 7) {
+        rot->x += (s16)this->unk_2CC;
+        rot->y += (s16)this->unk_2C8;
+        rot->z += (s16)this->unk_2C4;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6BF8.s")
+    return 0;
```

Comment:
```suggestion
    return false;
```
the `OverrideLimbDraw` callback returns a boolean indicating if the limb shouldnt be drawn (false = should be drawn)

Reply:
Oh ok, I was unaware of this, I'll keep it in mind for any future actors I do.